osbuild/util/ostree.py

```diff
@@ -214,7 +216,41 @@ def parse_input_commits(commits):
     return commits["path"], data["refs"]
 
 
-def deployment_path(root: PathLike, osname: str, ref: str, serial: int):
+def parse_deployment_option(root, deployment):
+    """Parse the deployment option and return the osname, ref, and serial
+
+    The `deployment` arg contains the following sub fields:
+    - osname: Name of the stateroot used in the deployment (ie. fedora-coreos)
+    - ref: OStree ref to used for the deployment (ie. fedora/aarch64/coreos/next)
+    - serial: The deployment serial (ie. 0)
+    - default: Boolean to determine whether the default ostree deployment should be used
+    """
+
+    default_deployment = deployment.get("default", False)
+    if default_deployment:
+        filenames = glob.glob(os.path.join(root, 'ostree/deploy/*/deploy/*.0'), recursive=True)
+        if len(filenames) < 1:
+            raise ValueError("Could not find deployment")
+        if len(filenames) > 1:
+            raise ValueError("More than one deployment found")
+
+        # We pick up the osname, commit, and serial from the filesystem
+        # here. We'll return the detected commit as the ref in this
+        # since it's a valid substitute for all subsequent uses in
+        # the code base.
+        f = re.search("/ostree/deploy/(.*)/deploy/(.*)\\.([0-9])", filenames[0])
```

Comment:
(pedantry++) mypy will complain here as re.search() may return None (even though in practise that can not happen)